### PR TITLE
Triangular_Number

### LINKED ISSUE #1112: Triangular Numbers in C,C++,Java and python

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #1112 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

-Added  Triangular Numbers in C, C++, Java, and python

#### Languages Used:
- C
- C++
- Python
- Java 

#### Files Added:
- Triangular_Number.c
- Triangular_Number.cpp
- Triangular_Number.py
- Triangular_Number.java

### CODE REVIEW

**Comment by singh-shreya6** [comment]
@jainaman224 Please have a look.
The entire description, code as well as example has been exactly copied from https://www.geeksforgeeks.org/triangular-numbers/

**Comment by singh-shreya6** [comment]
@jainaman224 Please have a look.
The entire description, code as well as example has been exactly copied from https://www.geeksforgeeks.org/triangular-numbers/
Please don't violate the basic principles of Gssoc. Always try to learn, implement and explain some algorithms in your own words.

**Comment by somya-kapoor** [comment]
I have tried to change the code as much as possible but  only 1 formula is used so can't change completely.

**Comment by singh-shreya6** [comment]
> I have tried to change the code as much as possible but only 1 formula is used so can't change completely.

Sure the problem follows quadratic equations and that cannot be changed, but the example used as well as the description and explanation of the problem could be written in your own language.

File: Triangular_Number/Triangular_Number.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*A number is termed as triangular number if we can represent it in the form of triangular grid of
```
Comment (singh-shreya6) [inline Triangular_Number/Triangular_Number.c]: Please change the description of problem. Exactly the same thing is written on Geeks for Geeks. 
Also add more comments which illustrate the usage of this equation and highlight your understanding of the problem by doing so.

**Comment by somya-kapoor** [comment]
@singh-shreya6 Done Please review

File: Triangular_Number/Triangular_Number.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+'''*The nth triangular number is the number of dots in the triangular arrangement
```
Comment (singh-shreya6) [inline Triangular_Number/Triangular_Number.py]: Remove the * here.

File: Triangular_Number/Triangular_Number.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/*The nth triangular number is the number of dots in the triangular arrangement
```
Comment (singh-shreya6) [inline Triangular_Number/Triangular_Number.c]: Please mention the quadratic equation which provides the solution to this algorithm in the description section.

**Comment by somya-kapoor** [comment]
@singh-shreya6 Done. Please review

File: Triangular_Number/Triangular_Number.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/*The nth triangular number is the number of dots in the triangular arrangement 
+with n dots on a side, and is equal to the sum of the n natural numbers from 1 to n.
+We can check if a number is triangular number or not by solving a quadratic equation :
+n2 + n + (-2 * num) = 0
```
Comment (singh-shreya6) [inline Triangular_Number/Triangular_Number.cpp]: Give n^2 for power notation

**Comment by somya-kapoor** [comment]
@singh-shreya6 Done

**Pull request merged**